### PR TITLE
Feat/texture

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/image/core/raster_image.h [hunk 1]
```diff
@@ -81,5 +81,5 @@ namespace geode
     private:
         IMPLEMENTATION_MEMBER( impl_ );
     };
-    ALIAS_2D_AND_3D( RasterImage );
+    ALIAS_1D_AND_2D_AND_3D( RasterImage );
```
Comment (francoisbonneau) [inline include/geode/image/core/raster_image.h:84]: I am not sure to understand why 1D raster image may be usefull.

**Review by francoisbonneau — APPROVED** [review]

File: src/geode/image/core/raster.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+#include <geode/basic/bitsery_archive.h>
+#include <geode/basic/pimpl_impl.h>
+#include <geode/basic/private/array_impl.h>
+
+#include <geode/image/core/rgb_color.h>
+
+namespace geode
+{
+    template < index_t dimension >
+    class RasterImage< dimension >::Impl : public detail::ArrayImpl< dimension >
```
Comment (MelchiorSchuh) [inline src/geode/image/core/raster.cpp:35]: I think you forgot to delete this file after copying it to raster_image.cpp

File: tests/image/test-raster-image.cpp [hunk 3]
```diff
@@ -59,6 +59,9 @@ void test()
     geode::save_raster( raster, "test.og_img2d" );
     const auto reload = geode::load_raster< 2 >( "test.og_img2d" );
     test_raster( reload );
+    auto raster2 = std::move( raster );
+    OPENGEODE_EXCEPTION(
+        raster2.nb_cells() == 100, "[Test] Wrog number of cells after move" );
```
Comment (MelchiorSchuh) [inline tests/image/test-raster-image.cpp:64]: Wrong

**Review by MelchiorSchuh — APPROVED** [review]

File: include/geode/geometry/detail/bitsery_archive.h [hunk 4]
```diff
@@ -31,13 +31,52 @@ namespace geode
 {
     namespace detail
     {
+
+        template < typename Type, typename Serializer >
+        void register_inlinedvector( PContext& context, absl::string_view type )
+        {
+            AttributeManager::register_attribute_type<
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply include/geode/geometry/detail/bitsery_archive.h:38]: No, this needs a C++14 syntax

**Pull request merged**